NuRadioReco/modules/io/coreas/readCoREASDetector.py

```diff
@@ -0,0 +1,313 @@
+        self.logger = logging.getLogger('NuRadioReco.readCoREASDetector')
+
+    def begin(self, input_file, interp_lowfreq=30*units.MHz, interp_highfreq=1000*units.MHz, log_level=logging.INFO, debug=False):
+            
+        """
+        begin method
+        initialize readCoREAS module
+        Parameters
+        ----------
+        input_files: input files
+            list of coreas hdf5 files
+        interp_lowfreq: float (default = 30)
+            lower frequency for the bandpass filter in interpolation
+        interp_highfreq: float (default = 1000)
+            higher frequency for the bandpass filter in interpolation
+        """
+        self.__input_file = input_file
+        self.__corsika = h5py.File(input_file, "r")
+        self.__interp_lowfreq = interp_lowfreq
+        self.__interp_highfreq = interp_highfreq
+        self.__sampling_rate = 1. / (self.__corsika['CoREAS'].attrs['TimeResolution'] * units.second)
+        self.logger.setLevel(log_level)
+        self.debug = debug
+
+        self.initialize_efield_interpolator()
+
+    def initialize_efield_interpolator(self):
+        zenith, azimuth, magnetic_field_vector = coreas.get_angles(self.__corsika)
+        self.cs = cstrafo.cstrafo(zenith, azimuth, magnetic_field_vector)
+
+        obs_positions = []
+        electric_field_on_sky = []
+        for j_obs, observer in enumerate(self.__corsika['CoREAS']['observers'].values()):
+            obs_positions.append(np.array([-observer.attrs['position'][1], observer.attrs['position'][0], 0]) * units.cm)
+            efield = np.array([observer[()][:,0]*units.second,
+                                -observer[()][:,2]*conversion_fieldstrength_cgs_to_SI,
+                                observer[()][:,1]*conversion_fieldstrength_cgs_to_SI,
+                                observer[()][:,3]*conversion_fieldstrength_cgs_to_SI])
+            efield_geo = self.cs.transform_from_magnetic_to_geographic(efield[1:,:])
+            # convert coreas efield to NuRadio spherical coordinated eR, eTheta, ePhi (on sky)
+            efield_on_sky = self.cs.transform_from_ground_to_onsky(efield_geo)
+            # insert time column before efield values
+            electric_field_on_sky.append(np.insert(efield_on_sky.T, 0, efield[0,:], axis = 1))
+
+        # shape: (n_observers, n_samples, (time, eR, eTheta, ePhi))
+        electric_field_on_sky = np.array(electric_field_on_sky)
+        electric_field_r_theta_phi = electric_field_on_sky[:,:,1:]
+        coreas_dt = (self.__corsika['CoREAS'].attrs['TimeResolution'] * units.second)
+
+        obs_positions = np.array(obs_positions)
+        # second to last dimension has to be 3 for the transformation
+        obs_positions_geo = self.cs.transform_from_magnetic_to_geographic(obs_positions.T)
+        # transforms the coreas observer positions into the vxB, vxvxB shower plane
+        obs_positions_vBvvB = self.cs.transform_to_vxB_vxvxB(obs_positions_geo).T
+
```

Comment:
In line to my earlier comments, I would initalize the pulses on the star-shaped pattern as `sim_stations` with `make_sim_station`. One does not have to add the stations to the shower/event but it this way you reduce code (copied from other files)

Reply:
I see what you mean. I left it here because we might want to change the on sky polarisation to handle to interpolation correctly.